### PR TITLE
Add treeview grouping to database objects

### PR DESCRIPTION
**Pull request by sm15455**
Things I have done:

- Added an ObjectType Views.
- Edited reverse engineering code to detect views
- Created a ViewModel that handles database objects treeview
- Created a ViewModel that handles root nodes ("table", "Views", "Stored procedures", "Functions")

Some methods in PickTablesViewModels now redirect to the first ViewModel.

### CODE REVIEW

**Comment by ErikEJ** [comment]
Wauw, thats a lot of changes!

How are tables without primary keys displayed?

File: src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml.cs [hunk 1]
```diff
@@ -58,5 +59,10 @@ IPickTablesDialog IPickTablesDialog.PreselectTables(IEnumerable<SerializationTab
             _selectTables(tables);
             return this;
         }
+
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/Dialogs/PickTablesDialog.xaml.cs:62]: Was this intentional?

File: src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs [hunk 2]
```diff
@@ -60,51 +55,36 @@ public string SearchText
...
-        {
-            foreach (var t in Tables)
-                PredefineSelection(t);
-        }
-
         private void Ok_Executed()
         {
-            foreach (var t in Tables)
-                t.PropertyChanged -= TableViewModel_PropertyChanged;
+            //foreach (var t in Tables)
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs:71]: Remove ?

File: src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs [hunk 3]
```diff
@@ -60,51 +55,36 @@ public string SearchText
...
         /// Currently at least a single table must be selected.
         /// </summary>
-        private bool Ok_CanExecute() => Tables.Any(m => m.IsSelected && !m.IsProcedure);
+        private bool Ok_CanExecute() => ObjectTree.Types.SelectMany(c => c.Objects).Any(m => m.IsSelected && !m.IsProcedure);
 
         private void Cancel_Executed()
         {
-            foreach (var t in Tables)
-                t.PropertyChanged -= TableViewModel_PropertyChanged;
+            //foreach (var t in Tables)
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs:84]: Remove?

File: src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs [hunk 4]
```diff
@@ -60,51 +55,36 @@ public string SearchText
...
+            //    t.PropertyChanged -= TableViewModel_PropertyChanged;
 
-            Tables.Clear();
+            //Tables.Clear();
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools/ViewModels/PickTablesViewModel.cs:87]: Remove

**Comment by sm15455** [comment]
That's not changed. The icon with the yellow alert triangle.

File: src/GUI/RevEng.Shared/ObjectType.cs [hunk 5]
```diff
@@ -5,5 +5,6 @@ public enum ObjectType
         Table = 0,
         Procedure = 1,
         ScalarFunction = 2,
+        View
```
Comment (ErikEJ) [inline src/GUI/RevEng.Shared/ObjectType.cs:8]: View = 3,

**Review by ErikEJ — CHANGES_REQUESTED** [review]

**Comment by ErikEJ** [comment]
Looks good, only test update is missing now

**Comment by sm15455** [comment]
I have fixed all tests and created new ones. I have tested UI and couldn't find bugs nor reproduce #612 .

**Review by ErikEJ — APPROVED** [review]

**Comment by ErikEJ** [comment]
Fantastic! I will update the command line tool asap

**Comment by ErikEJ** [comment]
Hmmm... Getting a blank screen with no table after getting latest master, and updating tool? @sm15455

**Comment by sm15455** [comment]
I'm investigating. I'll fix as I finish debugging.

**Comment by ErikEJ** [comment]
Looks like some data binding issue, it just shows a blank screen here, and the tables dto is populated ok:

    var pickTablesResult = ptd.ShowAndAwaitUserResponse(true);

**Comment by sm15455** [comment]
Committed fix to this branch. If it's a problem, I'll create a new PR.

**Comment by ErikEJ** [comment]
This is already merged, please create new PR, thanks

**Pull request merged**